command/agent/auth/token-file/token_file_test.go

```diff
@@ -0,0 +1,145 @@
+package token_file
+
+import (
+	"os"
+	"testing"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/command/agent/auth"
+	"github.com/hashicorp/vault/sdk/helper/logging"
+)
+
+func TestNewTokenFileAuthMethodEmptyConfig(t *testing.T) {
+	logger := logging.NewVaultLogger(log.Trace)
+	_, err := NewTokenFileAuthMethod(&auth.AuthConfig{
+		Logger: logger.Named("auth.method"),
+		Config: map[string]interface{}{},
+	})
+	if err == nil {
+		t.Fatal("Expected error due to empty config")
+	}
+}
+
+func TestNewTokenFileRemoveTokenString(t *testing.T) {
+	logger := logging.NewVaultLogger(log.Trace)
+	_, err := NewTokenFileAuthMethod(&auth.AuthConfig{
+		Logger: logger.Named("auth.method"),
+		Config: map[string]interface{}{
+			"token_file_path":                 "/some/path",
+			"remove_token_file_after_reading": "string",
+		},
+	})
+	if err == nil {
+		t.Fatalf("Expected error when giving string for bool")
+	}
+}
+
+func TestNewTokenFileEmptyFilePath(t *testing.T) {
+	logger := logging.NewVaultLogger(log.Trace)
+	_, err := NewTokenFileAuthMethod(&auth.AuthConfig{
+		Logger: logger.Named("auth.method"),
+		Config: map[string]interface{}{
+			"token_file_path": "",
+		},
+	})
+	if err == nil {
+		t.Fatalf("Expected error when giving empty file path")
+	}
+}
+
+func TestNewTokenFileAuthenticate(t *testing.T) {
+	tokenFile, err := os.CreateTemp("", "token_file")
```

Comment:
I think it's nicer these days to use t.TempDir.

Reply:
Good catch! Thanks for the suggestion, I'll update as approrpriate.